src/main/kotlin/nl/avisi/structurizr/site/generatr/site/SiteGenerator.kt

```diff
@@ -43,17 +42,30 @@ fun generateSite(
     assetsDir: File?,
     exportDir: File,
     branches: List<String>,
-    currentBranch: String
+    currentBranch: String,
+    serving: Boolean = false
 ) {
-    val generatorContext = GeneratorContext(version, workspace, branches, currentBranch) { key, url ->
+    val generatorContext = GeneratorContext(version, workspace, branches, currentBranch, serving) { key, url ->
         val view = workspace.views.views.single { view -> view.key == key }
         generateDiagramWithElementLinks(view, url, exportDir)
     }
 
+    deleteOldHashes(exportDir)
     if (assetsDir != null) copyAssets(assetsDir, File(exportDir, currentBranch))
     generateHtmlFiles(generatorContext, exportDir)
 }
 
+private fun deleteOldHashes(exportDir: File) = exportDir.walk().filter { it.extension == "md5" }
+    .forEach { it.delete() }
+
+private fun copySiteWideAsset(exportDir: File, name: String, extension: String) {
```

Comment:
Why do you have separate parameters for `name` and `extension`, and not just one `fileName` parameter, containing a full filename?